extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/client/MutinyVertxVaultClient.java

```diff
@@ -268,129 +266,133 @@ public VaultWrapResult wrap(String token, long ttl, Object object) {
                 throwVaultException(response);
             }
-            String jsonBody = response.body().string();
+            String jsonBody = response.body().toString();
             return resultClass == null ? null : mapper.readValue(jsonBody, resultClass);
-        } catch (IOException e) {
+        } catch (Exception e) {
             throw new VaultException(e);
         }
     }
 
-    private int exec(Request request) {
-        try (Response response = client.newCall(request).execute()) {
-            return response.code();
-        } catch (IOException e) {
+    private int exec(HttpRequest<Buffer> request) {
+        try {
+            return request.sendAndAwait().statusCode();
+        } catch (Exception e) {
             throw new VaultException(e);
         }
     }
 
-    private void throwVaultException(Response response) {
+    private void throwVaultException(HttpResponse<Buffer> response) {
         String body = null;
         try {
-            body = response.body().string();
+            body = response.body().toString();
         } catch (Exception e) {
             // ignore
         }
-        throw new VaultClientException(response.code(), body);
+        throw new VaultClientException(response.statusCode(), body);
     }
 
-    private Request.Builder builder(String path, String token) {
-        Request.Builder builder = new Request.Builder().url(getUrl(path));
+    private HttpRequest<Buffer> builder(String path, String token) {
+        HttpRequest<Buffer> request = builder(path);
         if (token != null) {
-            builder.header(X_VAULT_TOKEN, token);
+            request.putHeader(X_VAULT_TOKEN, token);
         }
-        return builder;
+        return request;
     }
 
-    private Request.Builder builder(String path) {
-        Request.Builder builder = new Request.Builder().url(getUrl(path));
-        return builder;
+    private HttpRequest<Buffer> builder(String path) {
+        return client.get(getUrl(path).toString());
     }
 
-    private Request.Builder builder(String path, Map<String, String> queryParams) {
-        HttpUrl.Builder httpBuilder = HttpUrl.parse(getUrl(path).toExternalForm()).newBuilder();
+    private HttpRequest<Buffer> builder(String path, Map<String, String> queryParams) {
+        HttpRequest<Buffer> request = builder(path);
         if (queryParams != null) {
-            queryParams.forEach((name, value) -> httpBuilder.addQueryParameter(name, value));
+            queryParams.forEach((name, value) -> request.addQueryParam(name, value));
         }
-        Request.Builder builder = new Request.Builder().url(httpBuilder.build());
-        return builder;
+        return request;
     }
 
-    private RequestBody requestBody(Object body) {
+    private String requestBody(Object body) {
         try {
-            return RequestBody.create(JSON, mapper.writeValueAsString(body));
+            return mapper.writeValueAsString(body);
```

Comment:
Pretty sure we can save the mapper, if we start using (Vert.x) JsonObject. It will use a mapper internally, but it avoids having another one.